src/Core/src/Layouts/LayoutExtensions.cs

```diff
@@ -83,6 +88,11 @@ static double AlignHorizontal(IView view, Rect bounds, Thickness margin)
 				// If the width is not set, we use the minimum between the MaxWidth or the bound's width
 				desiredWidth = IsExplicitSet(view.Width) ? desiredWidth : Math.Min(bounds.Width, view.MaximumWidth);
 			}
+			else if (!IsExplicitSet(view.Width))
```

Comment:
Given that `IsExplicitSet` is computed three times, we should probably create a variable at the beginning of the method:

```cs
var isWidthExplicitlySet = IsExplicitSet(view.Width);
```